ocpp1.6/central_system.go

```diff
@@ -40,7 +42,18 @@ func newCentralSystem(server *ocppj.Server) centralSystem {
 
 func (cs *centralSystem) error(err error) {
 	if cs.errC != nil {
-		cs.errC <- err
+		// It can happen that the error channel is getting closed
+		// when the central system is shutting down.
+		// If this happens right before this call here, we have a closed channel.
+		// To prevent a panic, we check if it is closed before sending the error.
+		cs.errCLock.Lock()
+		defer cs.errCLock.Unlock()
+		select {
+		case <-cs.errC:
+			// channel is closed, don't send the error
+		default:
+			cs.errC <- err
```

Comment:
Generally it's problematic if you have a Goroutine closing a channel that another Goroutine can write to. Not too familiar with the rest of the code anymore, but is there any reason we *need* to close `errC` at all? It's mostly used for logging, right? Then I don't see a reason why we can't just keep it open. Writers could just do a `select` and throw the error if the reader Goroutine already stopped.

Reply:
good idea 🤔 I'm also not familiar with it, I just fixed the issues :D 
how would the writer know if the readers goroutine has stopped?

Reply:
The write to the channel would block, but you can just wrap it in a `select` with a `default` case to prevent it from causing an issue